doc/Language/classtut.pod6

```diff
@@ -5,86 +5,60 @@
+the rare corner cases.
 
-Raku has a rich built-in syntax for defining and using classes.
+=head1 A quick overview
 
-A default constructor allows the setting of attributes for the created
-object:
+Let's start with an example to give an overview:
 
 =begin code
-class Point {
-    has Int $.x;
-    has Int $.y;
-}
-
 class Rectangle {
-    has Point $.lower;
-    has Point $.upper;
+    has Int $.a = 1;
+    has Int $.b = 1;
+
+    submethod TWEAK(:$log) {
+        say "Constructing Rectangle ($!a|$!b)" if $log;
+    }
 
     method area(--> Int) {
-        ($!upper.x - $!lower.x) * ( $!upper.y - $!lower.y);
+        return $!a * $!b;
     }
 }
 
-# Create a new Rectangle from two Points
-my $r = Rectangle.new(lower => Point.new(x => 0, y => 0),
-                      upper => Point.new(x => 10, y => 10));
+my $r1 = Rectangle.new(a => 2, b => 3);
+say $r1.area(); # OUTPUT: «6␤»
 
-say $r.area(); # OUTPUT: «100␤»
+my $r2 = Rectangle.new(b => 2, :log); # OUTPUT: «Constructing Rectangle (1|2)␤»
 =end code
 
-In the two classes, the default constructor is being used. This
-constructor will use named parameters in its invocation: C«Point.new(x =>
-0, y => 0)».
+We define a new I<Rectangle> class using the L<class|/language/objects#Classes> keyword. It has two I<attributes>, C<$!a> and C<$!b> introduced with the C<has> keyword. Both default to C<1>. Read only accessor methods are automatically generated. (Note the C<.> instead of C<!> in the declaration, which triggers the generation. Mnemonic: C<!> resembles a closed door, C<.> an open one.) 
 
-You can also provide your own constructor and C<BUILD> implementation.
-You need to do it for cases in which there are private attributes that
-might need to be populated during object construction, as in the
-example below:
+The L<method|/language/objects#Methods> named C<area> will return the area of the rectangle.
 
-=begin code
-# Example taken from
-# https://medium.freecodecamp.org/a-short-overview-of-object-oriented-software-design-c7aa0a622c83
-class Hero {
-    has @!inventory;
-    has Str $.name;
-    submethod BUILD( :$name, :@inventory ) {
-        $!name = $name;
-        @!inventory = @inventory
-    }
+Using the C<TWEAK> submethod, which is automatically called in the construction process we optionally log that a rectangle was created.
 
-    method act {
-        return @!inventory.pick;
-    }
-}
-my $hero = Hero.new(:name('Þor'),
-                    :inventory(['Mjölnir','Chariot','Bilskirnir']));
-say $hero.act;
-=end code
+It is rarely necessary to explicitly write a constructor. An automatically inherited default constructor called L<new|/language/objects#Object_construction> will automatically initialize attributes from named parameters passed to the constructor.
 
-In this case, we I<encapsulate> the private attribute C<@!inventory>; but
-private instance variables cannot be set by the default constructor, which is
-why we add a C<BUILD> submethod that takes care of that.
```

Comment:
Moved. See corresponding comment and https://github.com/Raku/doc/commit/0571bb9210aec7c25f111d715569b99878b4fa82.